common/src/main/java/dev/aurelium/auraskills/common/source/SourceLoader.java

```diff
@@ -194,6 +201,10 @@ public Map<SourceTag, List<XpSource>> loadTags(Skill skill, File contentDirector
         }
     }
 
+    private Collection<XpSource> getFilteredSource(List<XpSource> sources, String filter) {
+        return sources.stream().filter(source -> source.name().toLowerCase().contains(filter.replace("!", "").replace("*", "").toLowerCase())).collect(Collectors.toList());
```

Comment:
There's a potential issue in the logic for using `contains` to check the filter. For example, a filter of `*_sand` intended to catch `red_sand` and `soul_sand` will have the unintended effect of also including `red_sandstone`, even though there is no wildcard at the end.

Since your code only allows wildcards at the beginning and end of the name, we can change the logic to the following:
- If beginning wildcard, use `.endsWith`
- If ending wildcard, use `.startsWith`
- If both, use `.contains` like you do currently

Separately as a formatting thing, try to use line breaks to split very longs over about ~180 characters (you can set this as a visual guide in IntelliJ code style).